src/main/scala/cromwell/binding/values/WdlFile.scala

```diff
@@ -9,23 +9,38 @@ object WdlFile {
     if (path1.endsWith("/") || path2.startsWith("/")) path1 + path2
     else path1 + "/" + path2
   }
+
+  def apply(value: String, isGlob: Boolean = false): WdlFile = if (isGlob) WdlGlobFile(value) else WdlSingleFile(value)
 }
 
-case class WdlFile(value: String) extends WdlPrimitive {
+sealed trait WdlFile extends WdlPrimitive {
+  val value: String
 
   val wdlType: WdlType = WdlFileType
```

Comment:
I've followed Scott's advice on this - I don't understand the WdlType system well enough to comment. I think of this type as being something like a "File (de)Localisation Mapping" in which case you only get one mapping per File/Glob